edgelet/Makefile

```diff
@@ -123,7 +125,7 @@ rpm:
 install: release
 	$(INSTALL_PROGRAM) -D $(TARGET)/iotedged $(DESTDIR)$(bindir)/iotedged
 	$(INSTALL_PROGRAM) -D $(TARGET)/iotedge $(DESTDIR)$(bindir)/iotedge
-	$(INSTALL_DATA) -D $(srcdir)/contrib/config/linux/config.yaml $(DESTDIR)$(sysconfdir)/iotedge/config.yaml
+	$(INSTALL_DATA) -D $(srcdir)/contrib/config/linux/$(CONFIG_YAML) $(DESTDIR)$(sysconfdir)/iotedge/$(CONFIG_YAML)
```

Comment:
We should do the same thing for deb and rpm. So the rpm should also have the `.yaml.tempate` and  copy it to `.yaml` if that doesn't already exist.

Reply:
Could you help me a bit with RPM changes? I haven't done RPM packaging

Reply:
The equivalent place where you need to insert the "if `config.yaml` doesn't exist then copy `config.yaml.template` to it" line is around https://github.com/Azure/iotedge/blob/a85430e98c8fa436d8578fdcfa6d921f15ef1423/edgelet/contrib/centos/iotedge.spec#L74

To test it, you can build an rpm using `package.sh` with `PACKAGE_OS=centos7`, then install it in a `centos:7` Docker container to verify.

Reply:
> The equivalent place where you need to insert the "if `config.yaml` doesn't exist then copy `config.yaml.template` to it" line is around
> 
> https://github.com/Azure/iotedge/blob/a85430e98c8fa436d8578fdcfa6d921f15ef1423/edgelet/contrib/centos/iotedge.spec#L74
> 
> To test it, you can build an rpm using `package.sh` with `PACKAGE_OS=centos7`, then install it in a `centos:7` Docker container to verify.

While trying to test the RPM changes, during `PACKAGE_OS=centos7 PACKAGE_ARCH=amd64 ./edgelet/build/linux/package.sh`, the build process gave me an error that is unrelated to my changes

```
<snip>
[ 68%] Built target aziotsharedutil
Scanning dependencies of target utpm
[ 69%] [ 71%] [ 72%] [ 73%] [ 75%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/Marshal.c.o
Building C object deps/utpm/CMakeFiles/utpm.dir/src/tpm_codec.c.o
Building C object deps/utpm/CMakeFiles/utpm.dir/src/Memory.c.o
Building C object deps/utpm/CMakeFiles/utpm.dir/src/gbfiledescript.c.o
[ 76%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/tpm_socket_comm.c.o
Building C object deps/utpm/CMakeFiles/utpm.dir/src/tpm_comm_linux.c.o
Linking C static library libutpm.a
[ 76%] Built target utpm
Scanning dependencies of target iothsm
[ 77%] [ 78%] [ 80%] [ 81%] [ 82%] Building C object CMakeFiles/iothsm.dir/src/certificate_info.c.o
Building C object CMakeFiles/iothsm.dir/src/constants.c.o
Building C object CMakeFiles/iothsm.dir/src/edge_hsm_client_crypto.c.o
Building C object CMakeFiles/iothsm.dir/src/edge_enc_openssl_key.c.o
Building C object CMakeFiles/iothsm.dir/src/edge_hsm_client_store.c.o
[ 84%] Building C object CMakeFiles/iothsm.dir/src/edge_hsm_client_x509.c.o
[ 85%] [ 86%] Building C object CMakeFiles/iothsm.dir/src/edge_hsm_key_interface.c.o
Building C object CMakeFiles/iothsm.dir/src/edge_openssl_common.c.o
[ 88%] [ 89%] Building C object CMakeFiles/iothsm.dir/src/edge_sas_perform_sign_with_key.c.o
Building C object CMakeFiles/iothsm.dir/src/edge_pki_openssl.c.o
[ 90%] [ 92%] Building C object CMakeFiles/iothsm.dir/src/edge_sas_key.c.o
[ 93%] [ 94%] [ 96%] Building C object CMakeFiles/iothsm.dir/src/hsm_certificate_props.c.o
Building C object CMakeFiles/iothsm.dir/src/hsm_client_tpm_in_mem.c.o
Building C object CMakeFiles/iothsm.dir/src/hsm_client_data.c.o
Building C object CMakeFiles/iothsm.dir/src/hsm_client_tpm_device.c.o
[ 97%] Building C object CMakeFiles/iothsm.dir/src/hsm_client_tpm_select.c.o
make[2]: *** No rule to make target `/usr/lib/x86_64-linux-gnu/libssl.so', needed by `libiothsm.so.1.0.8'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 98%] [100%] Building C object CMakeFiles/iothsm.dir/src/hsm_log.c.o
Building C object CMakeFiles/iothsm.dir/src/hsm_utils.c.o
make[1]: *** [CMakeFiles/iothsm.dir/all] Error 2
make: *** [all] Error 2
```

Is there some other step I need before running package.sh?

Reply:
>`/usr/lib/x86_64-linux-gnu/libssl.so`

It's reusing the makefile from an ubuntu build. Make sure to `git clean` to remove the output of the ubuntu build first.

Reply:
> It's reusing the makefile from an ubuntu build. Make sure to `git clean` to remove the output of the ubuntu build first.

Thanks for the clarification. This is something else to add to Rust devguide along with the other information we discussed, which was
- git clone --recurse-submodules -j4 https://github.com/Azure/iotedge.git
- cd iotedge
- PACKAGE_OS=ubuntu18.04 PACKAGE_ARCH=amd64 ./edgelet/build/linux/package.sh

Reply:
(Decided against doing this for the rpm package in the main discussion thread.)